src/lib/esnext.object.d.ts

```diff
@@ -0,0 +1,13 @@
+interface ObjectConstructor {
+    /**
+     * Returns an object from an array of properties key/values
+     * @param entries Array that contains the properties key and value.
+     */
+    fromEntries<T>(entries: ArrayLike<[string, T]> | Iterable<[string, T]>): Record<string, T>;
```

Comment:
Would be nice to support tuple types but not sure the compiler is clever enough to deal with that without having to define 200 overloads

```ts
fromEntries<K1 extends string, V1>(entries: [[K1, V1]]): { [K1]: V1 }
fromEntries<K1 extends string, V1, K2 extends string, V2>(entries: [[K1, V1], [K2, V2]]): { [K1]: V1; [K2]: V2 }
// ...
```

Reply:
It was nice if typescript had any way to declare types mapping from tuple to object

Reply:
I have a use case for that :)
im doing a-lot of these manipulations.
and if i had a generic way to express it, that would be awesome

Reply:
@Bnaya can you share a realistic code snippet where object literal syntax doesn't work but fixed-length `fromEntries` does?